### DISCUSSION TITLE
Source of spacing between badges

### DISCUSSION THREAD

**Discussion by marceloverdijk** [twbs/bootstrap#39264] [Q&A] [answered]
I'm wondering which css property is responsible for the spacing between badges...

```
<div class="bd-example m-0 border-0">
  <span class="badge text-bg-primary">Primary</span>
  <span class="badge text-bg-secondary">Secondary</span>
  <span class="badge text-bg-success">Success</span>
  <span class="badge text-bg-danger">Danger</span>
  <span class="badge text-bg-warning">Warning</span>
  <span class="badge text-bg-info">Info</span>
  <span class="badge text-bg-light">Light</span>
  <span class="badge text-bg-dark">Dark</span>
</div>
```

<img width="784" alt="image" src="https://github.com/twbs/bootstrap/assets/197004/c4d425cd-b0c2-4149-8735-fa25cbb1b765">

As adding explicit 0 margin does not change anything to the rendering; e.g.:

```
  <span class="badge m-0 text-bg-primary">Primary</span>
  <span class="badge m-0 text-bg-secondary">Secondary</span>
```

It's not that I'm against the spacing but I have an issue with one of the official getbootstrap themes that the spacing is gone, which is undesired.

E.g. the theme from the demo page:

<img width="1129" alt="image" src="https://github.com/twbs/bootstrap/assets/197004/3914f7ce-1d0f-4dff-8957-883a0bbd27e4">

And I have this when I include just the scss from the theme in my own project:

<img width="1121" alt="image" src="https://github.com/twbs/bootstrap/assets/197004/fd47c995-4b14-4de7-8910-7396f7e0550f">

I tried disabling all css properties using the Chrome developer tools but I don't get the spacing away from the demo page.
If I could identify which property is responsible for the spacing, I could investigate better why in my project the spacing is gone.

**Comment by marceloverdijk** [comment]
I created another test project which does only use the default bootstrap.scss without modifications... and I got:

<img width="510" alt="image" src="https://github.com/twbs/bootstrap/assets/197004/4a612554-0e02-43fe-aece-2a2c41f2b131">

Also no spacing.
Could it be postcss related or something?

I have postcss config file like:

<img width="416" alt="image" src="https://github.com/twbs/bootstrap/assets/197004/84bfd16f-8b76-443c-946c-93b5aa89aff1">

**Comment by julien-deramond** [comment, marked as answer]
I think the spacing comes from whitespaces in the docs rendering:

![Screenshot 2023-10-03 at 08 26 07](https://github.com/twbs/bootstrap/assets/17381666/2b3eb685-f07d-4af9-9bed-f77174cd26ab)

Depending on the tools used to format the code, maybe they are removed in one case and not in another case.

**Reply by marceloverdijk** [reply]
That must be it. The theme demo itself indeed has also that line break/whitespace, and the page I rendered using Astro removes the line breaks so there is no whitespace.
I will add something like `me-1` or `me-2` to it.
